sdk/python/kfp/v2/compiler_cli_tests/compiler_cli_tests.py

```diff
@@ -116,6 +116,8 @@ def test_lightweight_python_functions_v2_with_outputs(self):
   def test_xgboost_sample_pipeline(self):
     self._test_compile_py_to_json('xgboost_sample_pipeline')
 
+  def test_pipeline_with_custom_job_spec(self):
+    self._test_compile_py_to_json('pipeline_with_custom_job_spec')
```

Comment:
Can we try to avoid snapshot tests in the future?
Snapshots are discouraged in testing, since they essentially freeze the implementation. Any small change to, say, lightweight component code generation will require changing all test data.

Let's try testing the relevant part. In this case that would be the compiler custom job specification.